vpr/src/device/rr_spatial_lookup.cpp

```diff
@@ -0,0 +1,108 @@
+/***************************************
+ * Methods for Object RRSpatialLookup
+ ***************************************/
+#include "vtr_assert.h"
+#include "rr_spatial_lookup.h"
+
+/****************************
+ * Constructors
+ ****************************/
+RRSpatialLookup::RRSpatialLookup(t_rr_node_indices& rr_node_indices)
+    : rr_node_indices_(rr_node_indices) {
+}
+
+/****************************
+ * Accessors
+ ****************************/
+RRNodeId RRSpatialLookup::find_node(int x,
+                                    int y,
+                                    t_rr_type type,
+                                    int ptc,
+                                    e_side side) const {
+    /* Find actual side to be used */
+    e_side node_side = side;
+    if (type == IPIN || type == OPIN) {
+        VTR_ASSERT_MSG(side != NUM_SIDES, "IPIN/OPIN must specify desired side (can not be default NUM_SIDES)");
+    } else {
+        VTR_ASSERT(type != IPIN && type != OPIN);
+        node_side = SIDES[0];
+    }
+
+    /* Currently need to swap x and y for CHANX because of chan, seg convention */
```

Comment:
_this_ is a place where I'd expect a more detailed comment, as this is clearly something that can surprise.

Reply:
Indeed. More explanation has been added.